inst/rmarkdown/templates/severity/skeleton/rmdchunks/id_key_data_linelist.Rmd

```diff
@@ -0,0 +1,65 @@
+## Identifying key data
+
+```{r define-key-variables, eval= params$data_columns == "Dates of onset and total deaths only", echo= params$data_columns == "Dates of onset and total deaths only"}
+# This code identifies key variables for analysis in the input dataset.
+group_var <- "Occupation"
+# Leave count_var as NULL if your data is really a linelist/patient-level data.
+# Update count_var to a character string with the name of the column that
+# contains case counts if your data is already aggregated.
+count_var <- NULL
+# Enter the geographical location where cases in your data took place
+country <- "Equatorial Guinea"
+cases <- "Onset_week"
+# Leave the parameter deaths below as it is if your data does not contain death dates
+deaths <- "deaths"
+# How many deaths in total are there in your data?
+total_fatalities <- 20
+```
+
+```{r define-key-variables2, eval= params$data_columns == "Dates of onset and death", echo= params$data_columns == "Dates of onset and death"}
+# This code identifies key variables for analysis in the input dataset.
+group_var <- "Occupation"
+# Leave count_var as NULL if your data is really a linelist/patient-level data.
+# Update count_var to a character string with the name of the column that
+# contains case counts if your data is already aggregated.
+count_var <- NULL
+# Enter the geographical location where cases in your data took place
+country <- "Equatorial Guinea"
+cases <- "Onset_week"
+deaths <- "Death_week"
+```
+
+
+```{r define-key-variables3, eval= params$data_columns == "Dates of onset and whether case died", echo= params$data_columns == "Dates of onset and whether case died"}
+# This code identifies key variables for analysis in the input dataset.
+group_var <- "transmission"
+# Leave count_var as NULL if your data is really a linelist/patient-level data.
+# Update count_var to a character string with the name of the column that
+# contains case counts if your data is already aggregated.
+count_var <- NULL
+# Enter the geographical location where cases in your data took place
+country <- "Equatorial Guinea"
+cases <- "disease_onset"
+# Indicate which column provides the date of outcome of the case
+outcome_date <- "disease_ended"
+# Indicate the column containing the case's outcome, e.g., "died" or "recovered"
+outcome_type <- "status"
+
+dat_raw <- dat_raw %>%
+  mutate(deaths = if_else(dat_raw[[outcome_type]] == "died", dat_raw[[outcome_date]], NA))
+
+deaths <- "deaths"
+```
```

Comment:
This seems to be almost equivalent to the 2nd case (with a one-line transformation done in L 48).

But it makes the template much more complicated. I think the convenience of doing this single step for the user is probably outweighed by the costs in terms of decreased usability and maintainability.

What do you think?